plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -100,7 +122,7 @@ private void refreshGrouping(List<ResourceGrouping> groupings, String kind) {
 			ConnectionsRegistry registry = (ConnectionsRegistry) parentElement;
 			return registry.getAll(Connection.class).toArray();
 		} else if (parentElement instanceof Connection) {
-			return ((Connection) parentElement).get(ResourceKind.PROJECT).toArray();
+			return ((Connection) parentElement).getResources(ResourceKind.PROJECT).toArray();
```

Comment:
this is a remoting call, are we sure that this wont be execute in the UI thread causing it to freeze?